### DISCUSSION TITLE
get value from nested attribute by directly using element/attribute path

### DISCUSSION THREAD

**Discussion by sesambrein83** [Hugovdberg/PIconnect#685] [Q&A] [answered]
Hi everyone, 

is it possible to get asset framework values directly by using the path of the element and path like this?

\\servername\ASD AF\R&D\Pilot Plants\H12\H12\H1|Oil Feed|Flow Control|PV

![image](https://user-images.githubusercontent.com/91729433/212715046-e26afa60-0c7e-418c-9d36-0b2bdd46993b.png)

i could do it by explicitly doing this in code like this:

![image](https://user-images.githubusercontent.com/91729433/212719356-436deba9-723c-4ebe-9911-5377f6c80629.png)

..but wouldn't it be neat if it could be done in one string?

**Comment by sesambrein83** [comment]
so for now i fixed it by splitting and looping. But there could be a more structured solution in the lib itself... maybe someone has a better idea?

`def getpointaverageAF(AFtag, startdatetime, enddatetime):

    with PI.PIAFDatabase() as db:        
        if len(AFtag) > 20 and "|" in AFtag:
            splittag = AFtag.split('|')
            elem = db.descendant(splittag[0])            
            finaltag = elem.attributes[splittag[1]]
            if len(splittag) > 2:
                for x in range(len(splittag)-2):
                    finaltag = finaltag.children[splittag[x+2]]
            result = finaltag.summary(startdatetime, enddatetime, SummaryType.AVERAGE).iloc[0]['AVERAGE']
        else:
            result = 'is not an AF Tag'
        return result

`print(getpointaverageAF(r'R&D\Pilot Plants\H12\H12\H1|Oil Feed|Flow Control|PV', '12-12-2022 12:00:00', '12-12-2022 13:00:00'))`

**Comment by Hugovdberg** [comment]
This is a nice addition that fits well with the way `PIAFDatabase.descendant` works. Just one question, is the `|` symbol also the official delimiter between attribute names or could that clash with valid attribute names?

**Reply by sesambrein83** [reply]
> This is a nice addition that fits well with the way `PIAFDatabase.descendant` works. Just one question, is the `|` symbol also the official delimiter between attribute names or could that clash with valid attribute names?

Thanks Hugo, Yes it is. When you click copy path on an attribute in PI Asset Framework explorer it copies the path in an \\element\element|attribute|attribute format

**Comment by Hugovdberg** [comment, marked as answer]
This has been added in the latest development version as `PIAFDatabase.search(query: str|list[str])`. It will probably be released this week in version 0.10.5

**Reply by sesambrein83** [reply]
Awesome thanks Hugo!

You might expect more features in the future. We are planning to use this plugin more at our production site.
